### PR TITLE
chore(releasing): Prepare v0.33.0 release

### PR DESCRIPTION
**Pull request by jszwedko**
Checklist: https://github.com/vectordotdev/vector/issues/18652

Also TODO:
- [x] Review added CSV encoder options for UX
- [x] Wait for additional PRs in https://github.com/vectordotdev/vector/milestone/115

### CODE REVIEW

File: website/content/en/highlights/2023-09-26-0-33-0-upgrade-guide.md [hunk 1]
```diff
@@ -48,15 +48,17 @@ Following our deprecation policy, v0.33.0 now disables the legacy OpenSSL provid
...
 
-
 ### Deprecations
 
 #### Default config location change {#default-config-location-change}
 
-The default config location `/etc/vector/vector.toml` which is used by Vector `0.32.0` is now deprecated. This location will still be used in `0.33.0`. The new default path is `/etc/vector/vector.yaml`, please migrate to this new default path or specify the config path explicitly.
-
-Vector `0.33.0` will attempt to load `/etc/vector/vector.toml` first, and if it is not present, it will fallback to `/etc/vector/vector.yaml`. However, Vector release `0.34.0` will automatically load `/etc/vector/vector.yaml` only.
+The default config location `/etc/vector/vector.toml` which is now deprecated. This location will
```
Comment (aliciascott) [inline website/content/en/highlights/2023-09-26-0-33-0-upgrade-guide.md:55]:
```suggestion
The default config location `/etc/vector/vector.toml` is now deprecated. This location will
```

File: website/content/en/highlights/2023-09-26-0-33-0-upgrade-guide.md [hunk 2]
```diff
@@ -66,13 +68,16 @@ package will be no longer be published beginning in the 0.34.0 release.
 
 #### Metadata field in the Vector protobuf definition {#vector-proto-metadata}
 
+You can ignore this if you aren't encoding or consuming events using Vector's native protobuf
```
Comment (aliciascott) [inline website/content/en/highlights/2023-09-26-0-33-0-upgrade-guide.md:71]:
```suggestion
Ignore if you aren't encoding or consuming events using Vector's native protobuf
```

**Review by aliciascott — APPROVED** [review]
A few small changes from docs side for wording, but otherwise approved!

File: website/cue/reference/releases/0.33.0.cue [hunk 3]
```diff
@@ -0,0 +1,621 @@
...
+				default location in the future rather than `/etc/vector/vector.toml`.
+				"""
+			pr_numbers: [18325, 18632, 18606, 18388, 18502, 18435, 18345, 18420, 18378]
+		},
+		{
+			type: "enhancement"
+			scopes: ["websocket sink"]
+			description: """
+				The `websocket` sink now accepts any data type that the configured codec accepts.
+				This means, for example, support logs, metrics, and traces when the `native` or
```
Comment (neuronull) [inline website/cue/reference/releases/0.33.0.cue:160]:
```suggestion
				For example, this means it supports logs, metrics, and traces when the `native` or
```

File: website/cue/reference/releases/0.33.0.cue [hunk 4]
```diff
@@ -0,0 +1,621 @@
...
+				"""
+			pr_numbers: [18609]
+			breaking: true
+		},
+		{
+			type: "enhancement"
+			scopes: ["datadog_agent source", "datadog_metrics sink"]
+			description: """
+				The `datadog_metrics` sink now transmits the origin of the metric to Datadog for display
+				in metrics explorer. Metrics coming the `datadog_agent` source have their origin passed
```
Comment (neuronull) [inline website/cue/reference/releases/0.33.0.cue:378]:
```suggestion
				in metrics explorer. Metrics from the `datadog_agent` source have their origin passed
```

File: website/cue/reference/releases/0.33.0.cue [hunk 5]
```diff
@@ -0,0 +1,621 @@
...
+				field to metadata rather than to the event itself.
+				"""
+			pr_numbers: [18259]
+		},
+		{
+			type: "enhancement"
+			scopes: ["config"]
+			description: """
+				Vector now parses more configuration fields that represent "event paths" (like the
+				`kafka` source `key_field`) at config parse time rather than at runtime. This
```
[the remark this answers is not in the document]
Comment (jszwedko) [reply website/cue/reference/releases/0.33.0.cue:107]: I'm actually not sure if _all_ are at this point or just most. Maybe @pront can confirm?

File: website/cue/reference/releases/0.33.0.cue [hunk 6]
```diff
@@ -0,0 +1,621 @@
...
+		{
+			type: "chore"
+			scopes: ["config"]
+			description: """
+				The default configuration language for Vector was updated from TOML to YAML. See the
+				[highlight](/highlights/2023-08-30-yaml-default-format) for more details on the
+				motivation for this change.
+
+				As part of this migration, Vector will prefer the `/etc/vector/vector.yaml` as the
+				default location in the future rather than `/etc/vector/vector.toml`.
```
[the remark this answers is not in the document]
Comment (jszwedko) [reply website/cue/reference/releases/0.33.0.cue:151]: Specifying a specific release has bitten us before when it hasn't actually made it into that release. It seems safer to just leave the door open to any future release.

**Review by jszwedko — COMMENTED** [review]
Thanks for the review @neuronull ! I accepted the changes and left a few comments in-line.

File: website/cue/reference/releases/0.33.0.cue (hunk 5 above)
[the remark this answers is not in the document]
Comment (pront) [reply website/cue/reference/releases/0.33.0.cue:107]: There are a few places where we still use `parse_path_and_insert` or `parse_target_path`. Perhaps a few of these can be refactored to use proper lookup. But most of them need to be parsed at runtime.

**Review by neuronull — APPROVED** [review]

File: website/cue/reference/releases/0.33.0.cue [hunk 7]
```diff
@@ -0,0 +1,691 @@
...
+
+				Features:
+
+				- `parse_key_value` can now parse values enclosed in single quote characters
+				- added `pretty` parameter for `encode_json` function to produce pretty-printed JSON string
+				- added `community_id` function for generation of [Community IDs](https://github.com/corelight/community-id-spec)
+				- `parse_aws_vpc_flow_log` can now handle VPC logs using version 5 fields
+				- the deprecated `to_timestamp` function was removed (use `parse_timestamp` and `from_unix_timestamp` instead)
+				"""
+			pr_numbers: [999999]
```
Comment (jszwedko) [inline website/cue/reference/releases/0.33.0.cue:454]: TODO: replace with actual PR number updating. I just put this placeholder here to fill out the other details in advance of the VRL upgrade.

File: website/cue/reference/releases/0.33.0.cue [hunk 8]
```diff
@@ -0,0 +1,691 @@
...
+		},
+		{
+			type: "fix"
+			scopes: ["kubernetes_logs source"]
+			description: """
+				Fix log event generation from the `kubernetes_logs` source when `log_namespacing` is
+				enabled. Previously it would emit empty log events.
+				"""
+			contributors: ["hhromic"]
+			pr_numbers: [18229]
```
Comment (hhromic) [inline website/cue/reference/releases/0.33.0.cue:82]: This wasn't me 🙈 :)
The PR number is also duplicated from the above entry.

Comment (jszwedko) [reply website/cue/reference/releases/0.33.0.cue:82]: Doh! Copy/paste error. Thanks for flagging.

File: website/cue/reference/releases/0.33.0.cue [hunk 9]
```diff
@@ -0,0 +1,714 @@
...
+releases: "0.33.0": {
+	date:     "2023-09-27"
+	codename: ""
+
+	whats_next: []
+
+	description: """
+		The Vector team is pleased to announce version 0.33.0!
+
+		Be sure to check out the [upgrade guide](/highlights/2023-09-26-0-33-0-upgrade-guide) for
```
Comment (dsmith3197) [inline website/cue/reference/releases/0.33.0.cue:12]: Here and below 🙂 
```suggestion
		Be sure to check out the [upgrade guide](/highlights/2023-09-27-0-33-0-upgrade-guide) for
```

**Review by dsmith3197 — APPROVED** [review]

File: website/cue/reference/releases/0.33.0.cue (hunk 9 above)
Comment (jszwedko) [reply website/cue/reference/releases/0.33.0.cue]: Doh, thanks!

**Review by pront — APPROVED** [review]

**Pull request merged**